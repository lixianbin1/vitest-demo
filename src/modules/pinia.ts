import { App } from 'vue'
import { createPinia } from 'pinia'

export const PiniaStore = createPinia()
export const install = (app:App) => {
  app.use(PiniaStore)
}
export default install