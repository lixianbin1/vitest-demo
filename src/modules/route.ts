import { App } from 'vue'
import { createRouter,createWebHashHistory } from 'vue-router'
import { setupLayouts } from 'virtual:generated-layouts'
import generatedRoutes from '~pages'

export const routes = setupLayouts(generatedRoutes)
export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
export const install = (app:App) => {
  app.use(router)
}
export default install