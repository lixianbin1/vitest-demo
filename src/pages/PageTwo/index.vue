<template>
  <div class="head">
    <h1>Page Two</h1>
  </div>
</template>

<script lang="ts" setup>

</script>

<style scoped>

</style>