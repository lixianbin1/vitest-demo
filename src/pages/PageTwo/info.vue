<template>
  <div class="head">
    <h1>Page Two Info</h1>
  </div>
</template>

<script lang="ts" setup>

</script>

<style scoped>

</style>