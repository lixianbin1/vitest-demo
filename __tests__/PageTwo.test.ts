import {expect,test,vi} from 'vitest'
import {render} from '@testing-library/vue'
import PageTwo from '../src/pages/PageTwo/index.vue'
import {i18n} from '../src/modules/i18n'
import {router} from '../src/modules/route'

test('PageTwo页面的快照',()=>{
  const { container } = render(PageTwo,{
    global: {
      plugins:[i18n,router]
    }
  })
  expect(container).toMatchSnapshot()
})