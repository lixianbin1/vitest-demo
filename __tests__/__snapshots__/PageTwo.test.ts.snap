// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTwo页面的快照 1`] = `
<div>
  <div
    class="head"
  >
    <h1>
      Page Two
    </h1>
  </div>
</div>
`;
